### DISCUSSION TITLE
Run cFS on older version of Linux

### DISCUSSION THREAD

**Discussion by technorogue321** [nasa/cFS#345] [Q&A] [answered]
My team and I were wanting to use cFS with an older version of Linux. Which version of cFS works with the Linux kernel 3.X?

**Comment by jphickey** [comment, marked as answer]
There aren't really any direct dependencies on a specific kernel version. CFS runs mostly in user space so it really has more to do with the C library and the user space API it provides.

The POSIX OSAL intentionally avoids using any bleeding-edge POSIX additions - but it does require/depend on some realtime additions.  Hopefully whatever C library you are using will provide sufficient POSIX APIs.  As long as the POSIX OSAL compiles and passes the tests on your platform, CFE/CFS should be fine.

See comment here for more info on how this relates to glibc environment (again, your flags may need to be tuned to the C library you are using):

https://github.com/nasa/osal/blob/ce11e7a20d4fd9176469659acb1ee3b371ba4291/src/bsp/generic-linux/CMakeLists.txt#L14-L26
